### ISSUE TITLE
Rule proposal: `vue/prefer-v-model`

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#2237]
**Please describe what the rule should do:**

This rule only applies to Vue 3: There, `:foo="bar" @update:foo="bar = $event"` can be simplified as `v-model:foo="bar". This rule suggests those simplifications.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- BAD -->
<my-component :foo="bar" @update:foo="bar = $event" />
<my-component v-bind:foo="bar" v-on:update:foo="bar = $event" />
<my-component :foo="bar" @update:foo="(foo) => bar = foo" />

<!-- GOOD -->
<my-component v-model:foo="bar" />
<my-component :foo="bar" @update:foo="baz = $event" /> <!-- different variables -->
<my-component :foo="bar" @update:foo="updateFoo($event)" /> <!-- function call -->
<my-component :foo="bar" @update:foo="(foo) => updateFoo(foo)" /> <!-- function call -->
```

**Additional context**

The rule is especially helpful after preparing a Vue 2→3 migration with renaming `model` props/event in a Vue 3-compatible manner, see also https://github.com/vuejs/eslint-plugin-vue/issues/2236.

**Comment by so1ve** [comment]
This is not safe.

https://vuejs.org/guide/essentials/forms.html
